lib/internal/streams/destroy.js

```diff
@@ -91,10 +92,10 @@ function _destroy(self, err, cb) {
   });
   if (result !== undefined && result !== null) {
     try {
-      const then = result.then;
-      if (typeof then === 'function') {
-        then.call(
-          result,
+      if (!SafeThenable)
+        SafeThenable = require('internal/per_context/safethenable');
+      new SafeThenable(result)
```

Comment:
Should wrap this in a `lazySafeThenable()` function to avoid the code duplication